core/src/test/scala/com/salesforce/op/ModelInsightsTest.scala

```diff
@@ -761,4 +774,29 @@ class ModelInsightsTest extends FlatSpec with PassengerSparkFixtureTest with Dou
     absError / bigCoeffSum < tol shouldBe true
     absError2 / smallCoeffSum < tol shouldBe true
   }
+
+  val tol2 = 0.1
+  val MomentsAndCard = getFeatureMomentsAndCard(standardizedLinpred, linRegDF._3)
+  val moments = MomentsAndCard._1
+  val cardinality = MomentsAndCard._2
+  val uniqueValsmallFeature = smallNorm.map(_.toDouble.get).toSet
+  val uniqueValbigFeature = bigNorm.map(_.toDouble.get).toSet
+
+  it should "correctly return moments calculation for numeric features" in {
```

Comment:
If you do the above suggestion, then you can also combine these two tests and make it so you're not hardcoding indices in your comparisons (downside is that since you have the feature name, you'd have to get the expected values from the dataframe itself):

```it should "correctly return moments calculation for numeric features" in {
    import spark.implicits._
    val df = linRegDF._3
    val featMap = getFeatureMomentMap(standardizedLinpred, linRegDF._3)

    // Go through each feature and check that the mean, variance, and unique counts match the data
    featMap.foreach(f => {
      f._2.moments.get.count shouldBe 1000

      val (expectedMean, expectedVariance) = df.select(avg(f._1), variance(f._1)).as[(Double, Double)]
        .collect().head
      math.abs((f._2.moments.get.mean - expectedMean) / expectedMean) < meanTol shouldBe true
      math.abs((f._2.moments.get.variance - expectedVariance) / expectedVariance) < varTol shouldBe true

      val actualUniques = df.select(f._1).as[Double].collect().toSet
      f._2.cardEstimate.get.valueCounts.keySet.map(_.toDouble).subsetOf(actualUniques) shouldBe true
    })
  }